pkg/loki/config_wrapper.go

```diff
@@ -136,6 +139,15 @@ func applyInstanceConfigs(r, defaults *ConfigWrapper) {
 		r.Ruler.Ring.InstanceInterfaceNames = r.Common.InstanceInterfaceNames
 		r.QueryScheduler.SchedulerRing.InstanceInterfaceNames = r.Common.InstanceInterfaceNames
 		r.Frontend.FrontendV2.InfNames = r.Common.InstanceInterfaceNames
+		r.IndexGateway.Ring.InstanceInterfaceNames = r.Common.InstanceInterfaceNames
+	}
+}
+
+// applyCommonReplicationFactor apply a common replication factor to the Ingester and Index Gateway rings.
+func applyCommonReplicationFactor(r, defaults *ConfigWrapper) {
+	if !reflect.DeepEqual(r.Common.ReplicationFactor, defaults.Common.ReplicationFactor) {
+		r.IndexGateway.Ring.ReplicationFactor = r.Common.ReplicationFactor
+		r.Ingester.LifecyclerConfig.RingConfig.ReplicationFactor = r.Common.ReplicationFactor
```

Comment:
Replication factor for ingester is being set at 3 places now. Let us drop it from here and `applyConfigToRings` since `applyIngesterReplicationFactor` does the final override.

Reply:
Good call, done.